### PR TITLE
fix(storage): build on MSVC+x86

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #8621

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8765)

### CODE REVIEW

File: google/cloud/storage/internal/object_requests.cc [hunk 1]
```diff
@@ -421,12 +421,14 @@ std::string UploadChunkRequest::RangeHeader() const {
...
     std::uint64_t new_offset) const {
   UploadChunkRequest result = *this;
-  if (new_offset < offset_) {
+  if (new_offset < offset_ || new_offset >= offset_ + payload_size()) {
     result.offset_ = new_offset;
     result.payload_.clear();
     return result;
   }
-  PopFrontBytes(result.payload_, new_offset - result.offset_);
+  // No chunk can be larger than what fits in memory, and thus this
```
Comment (devbww) [inline google/cloud/storage/internal/object_requests.cc:429]: "And thus this ..."?

Comment (coryan) [reply google/cloud/storage/internal/object_requests.cc]: Fixed.

File: google/cloud/storage/internal/object_requests.cc [hunk 2]
```diff
@@ -421,12 +421,15 @@ std::string UploadChunkRequest::RangeHeader() const {
...
   UploadChunkRequest result = *this;
-  if (new_offset < offset_) {
+  if (new_offset < offset_ || new_offset >= offset_ + payload_size()) {
     result.offset_ = new_offset;
     result.payload_.clear();
     return result;
   }
-  PopFrontBytes(result.payload_, new_offset - result.offset_);
+  // No chunk can be larger than what fits in memory, and because `new_offset`
+  // is the [offset_, offset_ + payload_size()], this static_cast<> is safe:
```
Comment (devbww) [inline google/cloud/storage/internal/object_requests.cc:430]: Nit: That interval is open on the end, so `)` would be a better closer.

s/is the/is in/

**Review by devbww — APPROVED** [review]

File: google/cloud/storage/internal/object_requests.cc (hunk 2 above)
Comment (devbww) [reply google/cloud/storage/internal/object_requests.cc:430]: Oops ... auto merge.

**Pull request merged**